analytics/lib/counts.py

```diff
@@ -14,21 +14,15 @@
 
 from collections import defaultdict, OrderedDict
 from datetime import timedelta, datetime
-import logging
+from zerver.lib.logging_util import create_logger
 import time
 
 ## Logging setup ##
 
-log_format = '%(asctime)s %(levelname)-8s %(message)s'
-logging.basicConfig(format=log_format)
-
-formatter = logging.Formatter(log_format)
-file_handler = logging.FileHandler(settings.ANALYTICS_LOG_PATH)
-file_handler.setFormatter(formatter)
-
-logger = logging.getLogger("zulip.management")
-logger.setLevel(logging.INFO)
-logger.addHandler(file_handler)
+logger = create_logger('%(asctime)s %(levelname)-8s %(message)s',
+                       'zulip.management',
+                       settings.ANALYTICS_LOG_PATH,
+                       'INFO')
```

Comment:
I think we should probably pass these arguments as kwargs, rather than positional arguments, for greater readability.

Reply:
Also, it appears that all the loggers use the same formatting function except this one, which is a bit more verbose in including the log level in the format string.  I think we might want to just make this format string be the default used by all of our loggers.

Since all of these specify the filename and log level, I think actually I like this as to way to call it:

create_logger(settings.ANALYTICS_LOG_PATH, "INFO", name="zulip.management")

and if we wanted to override the format, we'd do:

create_logger(settings.ANALYTICS_LOG_PATH, "INFO", name="zulip.management", format="%(foo)s...")